tools/logger/logger.c

```diff
@@ -171,6 +174,75 @@ static int append_filter_config(struct convert_config *config, const char *input
 	return 0;
 }
 
+/* Thanks to inotify, wait 5 minutes for 'expected_file' to appear in
+ * 'watched_dir'.  Then return fopen(expected_files). fopen() failures are NOT
+ * handled; e.g. NULL from fopen() is just passed through.
+ */
+static FILE *wait_fopen(const char *watched_dir, const char *expected_file)
+{
+	if (access(watched_dir, R_OK)) {
+		fprintf(stderr, "error: %s() cannot read %s\n", __func__, watched_dir);
+		return NULL;
+	}
+
+	const int iqueue = inotify_init();
+	const int dwatch = inotify_add_watch(iqueue, watched_dir, IN_CREATE);
+	FILE *fstream;
+
+	char * const fpath = malloc(strlen(watched_dir) + 1 + strlen(expected_file) + 1);
+
+	strcpy(fpath, watched_dir);
+	strcat(fpath, "/");
+	strcat(fpath, expected_file);
+
+	/* Not racy because the inotify watch was set first. */
+	if (!access(fpath, F_OK))
+		goto fopenit;
+
+	fprintf(stdout, "%s: waiting in %s/ for %s\n", APP_NAME, watched_dir, expected_file);
+	fflush(stdout);
+
+	while (1) {
+		char evlist[1 * sizeof(struct inotify_event)];
+		const int timeout_min = 5;
+		struct pollfd pfd[1] = {{ .fd = iqueue, .events = POLLIN, .revents = 0 }};
+		int pret = poll(pfd, sizeof(pfd) / sizeof(struct pollfd), timeout_min * 60 * 1000);
+
+		if (pret == 0) {
```

Comment:
Maybe have a define for the "5" minutes. Or maybe just wait indefinitely? I mean if someone depends on error being returned if driver doesn't show up, 5 minutes is an awful lot of time to wait for (e.g. for test automation). So it could might as well be infinite wait.

Reply:
This code started with an infinite wait but then I realized a serious backward-compatibility issue: some test code could already be trying to start the logger before the driver is loaded (and ignoring the error as usual in shell scripts or C).